src/main.rs

```diff
@@ -280,35 +243,56 @@ impl Inlyne {
                             if !scrollbar_held {
                                 scrollbar_held = true;
                             }
-                        } else if click_scheduled && !jumped {
+                        } else if self.renderer.selection.is_none() {
                             self.renderer.selection = Some((loc, loc));
-                        } else if let Some(ref mut selection) = self.renderer.selection {
+                        } else if let Some(selection) = &mut self.renderer.selection {
                             if mouse_down {
                                 selection.1 = loc;
                                 self.window.request_redraw();
                             }
                         }
-
-                        if !over_link {
-                            self.window.set_cursor_icon(CursorIcon::Default);
-                        }
-                        click_scheduled = false;
                     }
                     WindowEvent::MouseInput {
                         state,
                         button: MouseButton::Left,
                         ..
                     } => match state {
                         ElementState::Pressed => {
+                            // Try to click a link
+                            let screen_size = self.renderer.screen_size();
+                            if let Some(hoverable) = Self::find_hoverable(
+                                &self.elements,
+                                &mut self.renderer.glyph_brush,
+                                last_loc,
+                                screen_size,
+                                self.renderer.zoom,
+                            ) {
+                                let maybe_link = match hoverable {
+                                    Hoverable::Image(Image { is_link, .. }) => is_link,
+                                    Hoverable::Text(Text { link, .. }) => link,
+                                };
+
+                                if let Some(link) = maybe_link {
+                                    if open::that(link).is_err() {
+                                        if let Some(anchor_pos) =
+                                            self.renderer.positioner.anchors.get(link)
+                                        {
+                                            self.renderer.set_scroll_y(*anchor_pos);
+                                            self.window.request_redraw();
```

Comment:
This is the redundant redraw I mentioned. Redraws are by far the most expensive operation and I don't think this is needed.